backend/storage-service/src/jvmMain/kotlin/file/services/SynchronizationService.kt

```diff
@@ -0,0 +1,205 @@
+        return syncthing.config.devices.minByOrNull { device ->
+            db.withSession { session ->
+                session.sendPreparedStatement(
+                    {
+                        setParameter("device", device.id)
+                    },
+                    """
+                        select path
+                        from synchronized_folders
+                        where device_id = :device
+                    """
+                )
+            }.rows.sumOf {
+                CephFsFastDirectoryStats.getRecursiveSize(File(fsPath, it.getField(SynchronizedFoldersTable.path)))
+            }
+        }
+    }
+
+    suspend fun addFolder(actor: Actor, request: SynchronizationAddFolderRequest) {
+        // TODO Check for number of files in folder before adding (limit to 1000_000)
+        if (CephFsFastDirectoryStats.getRecursiveFileCount(File(fsPath, request.path)) > 1000_000) {
+            throw RPCException("Number of files in directory exceeded for synchronization", HttpStatusCode.Forbidden)
+        }
+
+        val id = UUID.randomUUID().toString()
+        val device = chooseFolderDevice()
+        val accessType = if (aclService.hasPermission(request.path, actor.username, AccessRight.WRITE)) {
+            SynchronizationType.SEND_RECEIVE
+        } else {
+            SynchronizationType.SEND_ONLY
+        }
+
+        if (device == null) {
+            throw RPCException.fromStatusCode(HttpStatusCode.InternalServerError)
+        }
+
+        db.withSession { session ->
+            session.sendPreparedStatement(
+                {
+                    setParameter("id", id)
+                    setParameter("device", device.id)
+                    setParameter("path", request.path)
+                    setParameter("user", actor.username)
+                    setParameter("access", accessType.name)
+                },
+                """
+                    insert into storage.synchronized_folders(
+                        id, 
+                        device_id, 
+                        path,
+                        user_id,
+                        access_type
+                    ) values (
+                        :id,
+                        :device,
+                        :path,
+                        :user,
+                        :access
+                    )
+                """
+            )
+        }
+
+        syncthing.writeConfig()
+    }
+
+    suspend fun removeFolder(actor: Actor, request: SynchronizationRemoveFolderRequest) {
+        db.withSession { session ->
+            session.sendPreparedStatement(
+                {
+                    setParameter("id", request.id)
+                    setParameter("user", actor.username)
+                },
+                """
+                    delete from storage.synchronized_folders
+                    where id = :id and user_id = :user
+                """
+            )
+        }
+
+        syncthing.writeConfig()
```

Comment:
Add a check to see if you really need to write configuration. For example if the code above didn't delete anything, for example in the case of lacking permissions, this will do nothing.